### ISSUE TITLE
Eloquent helper code not being generated for models with table name being written in runtime

### ISSUE THREAD

**Issue by lsalazarm99** [laravel-idea/plugin#278]
I'm working with a database I only read (I can write too but this project doesn't need it). The database was created for another project. It doesn't follow Laravel conventions. For example, the tables' name is all uppercase and snake case (for example, `PS_USERS`).

Given that, I use the same table name for the model name and, in runtime, I assign the name of the model to the `$table`.
```php
// {Database}Model.php - This is a parent class for all the models of that database. It extends Laravel's Model class.

public function __construct(array $attributes = [])
{
    parent::__construct($attributes);
    $this->table = static::getTableName();
}

public static function getTableName(): string
{
    $exploded = explode('\\', static::class);

    return end($exploded);
}
```

This works with Barry's package but not with Laravel IDEA Code Generation. I guess it is because Barry's one executes the PHP code so the `$table` variable gets assigned. Also, it doesn't need to run a code generation for each database I'm using because it already connects to all the specified in Laravel configuration.

Without specifying the table name in a `$table` variable explicitly, is there any way I can get the code generation for these models?

**Comment by adelf** [comment]
> I guess it is because Barry's one executes the PHP code so the $table variable gets assigned.

Exactly. Maybe in the future, I'll implement some combination with PHP+plugin to solve this...
